src/pip/_internal/commands/wheel.py

```diff
@@ -177,6 +187,23 @@ def run(self, options, args):
                         raise CommandError(
                             "Failed to build one or more wheels"
                         )
+                    if wb.path_to_wheelnames is not None:
+                        entries_to_save = wb.wheel_filenames[::]
```

Comment:
To avoid making ever larger, harder-to-test functions and to make the code more maintainable, I think everything inside this if block should be made into a separate, independently tested (unit-tested) function.